src/cc65/declare.c

```diff
@@ -517,21 +684,31 @@ static int ParseFieldWidth (Declaration* Decl)
     /* Read the width */
     NextToken ();
     ConstAbsIntExpr (hie1, &Expr);
+    if (!IsClassInt (Decl->Type)) {
+        /* Only integer types may be used for bit-fields */
+        Error ("Bit-field has invalid type ''");
```

Comment:
It would be nice to print the type while you're changing this.

Reply:
Strange mistake. Fixed.